mods/seasonal/moves.js

```diff
@@ -2648,10 +2649,10 @@ let BattleMovedex = {
 				return 5;
 			},
 			onBeforeMovePriority: 2,
-			onBeforeMove: function (pokemon, target, move) {
-				let hazards = ['reflect', 'lightscreen', 'auroraveil', 'safeguard', 'mist', 'spikes', 'toxicspikes', 'stealthrock', 'stickyweb', 'hazardpass', 'beskyttelsesnet', 'bringerofdarkness'];
-				if (hazards.includes(move.id)) {
-					this.add('-activate', target, 'move: Prismatic Terrain');
+			onBeforeMove: function (target, source, move) {
+				let hazardMoves = ['reflect', 'lightscreen', 'auroraveil', 'safeguard', 'mist', 'spikes', 'toxicspikes', 'stealthrock', 'stickyweb', 'hazardpass', 'beskyttelsesnet', 'bringerofdarkness', 'soulbend', 'smokebomb', 'hurl'];
+				if (hazardMoves.includes(move.id)) {
+					this.add('-message', `${source.name} couldn't use ${move.name} because of the prismatic terrain!`);
```

Comment:
i assume the activate message displays the blue box and says "Prismatic Terrain activiated!"?

Reply:
It says X's Prismatic Terrain Activated